packages/contracts/test/test-helpers/ovm-helpers.ts

```diff
@@ -0,0 +1,322 @@
+/* External Imports */
+import { ethers, Signer, Contract, ContractFactory } from 'ethers'
+import { Log, TransactionReceipt, JsonRpcProvider } from 'ethers/providers'
+import {
+  abi,
+  add0x,
+  hexStrToBuf,
+  getLogger,
+  logError,
+  BloomFilter,
+  numberToHexString,
+  bufToHexString,
+  getCurrentTime,
+} from '@eth-optimism/core-utils'
+
+/* Internal Imports */
+import { ZERO_ADDRESS, GAS_LIMIT } from './constants'
+import { Address } from './types'
+import { encodeMethodId, encodeRawArguments } from './ethereum-helpers'
+
+/* Contract Imports */
+import { getContractInterface } from '../../index'
+
+const ExecutionManagerInterface = getContractInterface('ExecutionManager')
+const logger = getLogger('contracts:test-helpers', true)
+const revertMessagePrefix: string =
+  'VM Exception while processing transaction: revert '
+
+/**
+ * Convert internal transaction logs into OVM logs. Or in other words, take the logs which
+ * are emitted by a normal Ganache or Geth node (this will include logs from the ExecutionManager),
+ * parse them, and then convert them into logs which look like they would if you were running this tx
+ * using an OVM backend.
+ *
+ * NOTE: The input logs MUST NOT be stripped of any Execution Manager events, or this function will break.
+ *
+ * @param logs An array of internal transaction logs which we will parse and then convert.
+ * @param executionManagerAddress The address of the Execution Manager contract for log parsing.
+ * @return the converted logs
+ */
+export const convertInternalLogsToOvmLogs = (
```

Comment:
At some point I believe this should be moved into the `rollup-fullnode` package (or deleted generally) cc @K-Ho